src/toil/common.py

```diff
@@ -84,6 +84,11 @@ def __init__(self):
         self.betaInertia = 1.2
         self.scaleInterval = 10
         self.preemptableCompensation = 0.0
+        
+        # Parameters to limit service jobs, so preventing deadlock scheduling scenarios
+        self.maxPreemptableServiceJobs = sys.maxint 
+        self.maxServiceJobs = sys.maxint
```

Comment:
Correct, I did not enable this by default because I was worried about it making weird side effects. I prefer this conservative approach until we have tested it at scale.